src/library_exceptions.js

```diff
@@ -393,6 +395,7 @@ var LibraryExceptions = {
     var thrownType = info.get_type();
     var catchInfo = new CatchInfo();
     catchInfo.set_base_ptr(thrown);
+    catchInfo.set_adjusted_ptr(thrown);
```

Comment:
I don't follow how this is NFC. This extra call will write a change to memory, and if we return on line 401 then it is noticeable, isn't it?

Reply:
Oh sorry, yes, this is setting the adjusted pointer in all cases,  I forgot I ended up adding this line.

Basically __cxa_can_catch expects the existing value here to be the base_ptr and then it updates it accordingly.

The `get_exception_ptr` method will always prefer the adjusted pointer and fall back to the base ptr if its zero.   I think in practice this now means that the fallback is not needed, but I also don't think there is any observable effect.    The adjusted pointer will be equal the base pointer in the case that no adjustment was made.

Reply:
In modifying this code I'm fairly sure we have enough test coverage of this adjustment in our test suite, so if this was not NFC I think we would have seen a test failure.

Reply:
I see. Seems likely to be correct, yes.